### PR TITLE
Port to Python 3

### PR DESCRIPTION
**Pull request by ayushnawal**

### CODE REVIEW

**Comment by ayushnawal** [comment]
@quozl please review!

**Comment by quozl** [comment]
Thanks.  Reviewed.  Tested.  Please fix the warning about `GObject.source_remove`.

**Comment by ayushnawal** [comment]
actually I tried, but I am not able to reproduce this warning, so I am not able to get and test any proper fix for it.

a little guidance can help me here.

Thanks :grin:

**Comment by chimosky** [comment]
@ayushnawal try accessing all the buttons in the toolbar. This is the error.

```
/home/ibiam/Activities/iknowmyabcs/page.py:202: Warning: Source ID 1524 was not found when attempting to remove it
  GObject.source_remove(self.timeout)
```

**Comment by ayushnawal** [comment]
@chimosky @quozl, please review

**Comment by quozl** [comment]
Thanks.  Reviewed.  Waiting on https://github.com/sugarlabs/iknowmyabcs/pull/11 per https://github.com/sugarlabs/sugar-docs/blob/master/src/python-porting-guide.md#how-to-port-an-activity-to-python-3 "_making sure the activity works properly before porting, closing any solved issues, merging any pull requests or branches ..._".  You can progress this by adding those commits.

**Comment by ayushnawal** [comment]
How can I proceed on #11 

Created account on http://activities.sugarlabs.org

I have very little idea on how to release an activity.

A little detailed guide will be much helpful for me to release this as well as many other activities.

**Comment by quozl** [comment]
You can progress this pull request by adding those commits from the other pull request, i.e. by rebasing your branch on the other.

You should not attempt to release a Python 3 activity to http://activities.sugarlabs.org/ as we use that now for Python 2 activities.  We have no app store yet for Python 3 activities.

For detail on release, I can't guess what you don't know, so see the guide to contributing, and observe other activity releases, mostly by me, on the mailing list.  If you can't tell what I've done to release, then ask specific questions.

**Comment by ayushnawal** [comment]
@quozl added commits from the other pull request as suggested.

**Comment by quozl** [comment]
Thanks.  NEWS does not fully describe the nature of the changes since previous version.

**Comment by ayushnawal** [comment]
Thanks, I didn't completely understand.

> NEWS does not fully describe the nature of the changes since the previous version.

by this are you saying to add the recent commits in the NEWS file or you meant to describe changes already mentioned in the file.

sorry, but I never worked on a NEWS file, want to learn.

**Comment by quozl** [comment]
> > NEWS does not fully describe the nature of the changes since the previous version.
> 
> by this are you saying to add the recent commits in the NEWS file or you meant to describe changes already mentioned in the file.

By this I'm saying that the NEWS file is missing changes.

> sorry, but I never worked on a NEWS file, want to learn.

Look at the NEWS files of each of the Fructose activities.

**Comment by ayushnawal** [comment]
I have added the recent commit of yours in `NEWS` as you suggested to add missing changes since the previous version.
@quozl, please review

**Comment by quozl** [comment]
Thanks.  I've reviewed this pull request in detail again.

* the [aggregate change](https://github.com/sugarlabs/iknowmyabcs/pull/12/files) which at present consists of one line added to NEWS, changing `exec`, porting of the source files `page.py` and `utils/sprites.py`, and fixing the source_remove warning,
* each incremental commit, one by one.

I couldn't see why some of the incremental commits weren't in the aggregate change, until I used `gitk` to visualise your branch;

gitk

It looks like your branch diverged from master a long time ago.  I don't know what happened to cause that.  I suggest you pull your master branch from our master branch.  I suggest you rebase your port branch from a master branch.

**Comment by ayushnawal** [comment]
rebased with the master branch as you suggested. 
it is following master now, with all commits as it should have been.

I think this happened because I rebased the branch with #11 as you suggested [here](https://github.com/sugarlabs/iknowmyabcs/pull/12#issuecomment-579552867) but those commits were already in the master branch.

File: utils/sprites.py [hunk 1]
```diff
@@ -251,7 +251,7 @@ def set_layer(self, layer=None):
     def set_label(self, new_label, i=0):
         ''' Set the label drawn on the sprite '''
         self._extend_labels_array(i)
-        if type(new_label) is str or type(new_label) is unicode:
+        if type(new_label) is str or type(new_label) is str:
```
Comment (quozl) [inline utils/sprites.py]: This line checks the same thing twice.  It is like `if False or False`, or `if True or True`.

Comment (quozl) [reply utils/sprites.py]: Also, other activities that contain sprites.py use isinstance instead of type; did you compare against them at all?

**Review by quozl — COMMENTED** [review]
Thanks.  Reviewed.

I guess you focused on this repository only?  In my repositories I've found 20 different versions of sprites.py, each with different bugs.  This will be the 21st.

**Comment by ayushnawal** [comment]
whoops .. I didn't notice that :sweat_smile: 

Fixed. Tested.

Although I didn't focus on `sprites.py` particularly in any other activity but can you please suggest me some of those, I will try to fix them.

Thanks

**Comment by quozl** [comment]
Next time, if you need to find `sprites.py`, try searching.

https://github.com/search?utf8=%E2%9C%93&q=org%3Asugarlabs+filename%3Asprites.py&type=Code&ref=advsearch&l=&l=

**Comment by ayushnawal** [comment]
@srevinsaju this PR is ready from my side.
* Port to Python3
* Fix source_remove warning
* update news 
* Fix `sprites.py` bug 

No any errors and warnings

**Comment by srevinsaju** [comment]
Thanks @ayushnawal, I will check it out when I have time, (my exams getting over on 20th)

**Comment by ayushnawal** [comment]
@quozl  please [see](https://github.com/sugarlabs/iknowmyabcs/pull/12#issuecomment-586987327)

**Comment by quozl** [comment]
Merged by hand.  https://github.com/sugarlabs/iknowmyabcs/commits/master

**Pull request closed without merging**